osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -68,16 +70,38 @@ public SampleChannel GetSample(ISampleInfo sampleInfo)
 
         public IBindable<TValue> GetConfig<TLookup, TValue>(TLookup lookup)
         {
-            if (AllowConfigurationLookup && skin != null)
+            if (skin != null)
             {
-                var bindable = skin.GetConfig<TLookup, TValue>(lookup);
-                if (bindable != null)
-                    return bindable;
+                switch (lookup)
+                {
+                    // todo: the GlobalSkinColours switch is pulled from LegacySkin and should not exist.
+                    // will likely change based on how databased storage of skin configuration goes.
+                    case GlobalSkinColours global:
+                        switch (global)
+                        {
+                            case GlobalSkinColours.ComboColours:
+                                return getBindable<TLookup, TValue>(lookup, AllowColourLookup);
+                        }
+
+                        break;
```

Comment:
not sure this is handling `GlobalSkinColours` lookups that aren't `ComboColours` correctly. right now they'll always get skipped in the inner switch, and therefore get the value from the fallback source.

this would be one of the rare cases where `case...when` syntax would be useful, to be able to do, say:

```csharp
switch (lookup)
{
	case GlobalSkinColours global when global == GlobalSkinColours.ComboColours:
		// do the skin colour thing

	default:
		// do the normal lookup thing
}
```

but I know @peppy has reservations against that syntax...

Reply:
The only other value for the `GlobalSkinColours` enum is `GlobalSkinColours.MenuGlow` which, based on what I've seen in the code, is only used for the main menu visualizer and main menu side flashes and not for any beatmap skin related settings/checks.

The code I originally looked at for this in `osu.Game.Skinning.DefaultSkin.cs` also skips that case:
```csharp
switch (lookup)
{
    // todo: this code is pulled from LegacySkin and should not exist.
    // will likely change based on how databased storage of skin configuration goes.
    case GlobalSkinColours global:
        switch (global)
        {
            case GlobalSkinColours.ComboColours:
                return SkinUtils.As<TValue>(new Bindable<IReadOnlyList<Color4>>(Configuration.ComboColours));
        }

        break;
}
```

Reply:
`DefaultSkin` is special in that regard as it doesn't really support `MenuGlow` (compare to `LegacySkin` which does). truth be told i'm not 100% sure that there will ever be any gameplay-relevant colour lookups that aren't combo colours, which is mainly why i'm bringing that up.

now that i think about it, something that should probably be checked is how catch is handling this right now as it has a couple of custom skin colours (related to hyperdashes).

Reply:
Ahh, I do see `LegacySkin` has a default in its switch statement, I didn't realize that `DefaultSkin` was a special case. I suppose I could either implement the `case...when` syntax you suggested above or include the default case in the inner switch as well.

I will check how Catch is handling the `CatchSkinColour` scenarios and go through the other rulesets to make sure I didn't miss any other colour settings, and of course will update/add any relevant tests as needed.

Reply:
> not sure this is handling `GlobalSkinColours` lookups that aren't `ComboColours` correctly. right now they'll always get skipped in the inner switch, and therefore get the value from the fallback source.
> 
> this would be one of the rare cases where `case...when` syntax would be useful, to be able to do, say:
> 
> ```cs
> switch (lookup)
> {
> 	case GlobalSkinColours global when global == GlobalSkinColours.ComboColours:
> 		// do the skin colour thing
> 
> 	default:
> 		// do the normal lookup thing
> }
> ```
> 
> but I know @peppy has reservations against that syntax...

Fortunately, I think I was able to find a better solution to this check when I was looking into handling the CTB hyper dash colouring.